prow/gerrit/source/source.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package source contains functions that help with Gerrit source control
+// specific logics.
+package source
+
+import (
+	"fmt"
+	"strings"
+)
+
+// IsGerritOrg tells whether the org is a Gerrit org or not. It returns true
+// when the org string starts with https://.
+func IsGerritOrg(org string) bool {
+	return strings.HasPrefix(org, "https://") || strings.HasPrefix(org, "http://")
+}
+
+// CloneURIFromOrgRepo returns normalized cloneURI from org and repo. The
+// returns cloneURI will always have https:// or http:// prefix, and there is no
+// trailing slash at the end.
+func CloneURIFromOrgRepo(org, repo string) string {
+	return NormalizeCloneURI(orgRepo(org, repo))
+}
+
+// NormalizeOrg returns normalized org. It ensures that org always has https://
+// or http:// prefix, and there is no trailing slash at the end. This function
+// should be used everywhere that Gerrit org is referenced.
+func NormalizeOrg(org string) string {
+	return ensuresHTTPSPrefix(org)
+}
+
+// NormalizeCloneURI returns normalized cloneURI. It ensures that cloneURI
+// always has https:// or http:// prefix, and there is no trailing slash at the
+// end. This function should be used everywhere that Gerrit cloneURI is
+// referenced.
+func NormalizeCloneURI(cloneURI string) string {
+	return ensuresHTTPSPrefix(strings.TrimRight(cloneURI, "/"))
```

Comment:
nit: The `strings.TrimRight` is redundant and including it here, but not for `NormalizeOrg` was a bit confusing at first.

Reply:
good catch, I'd like these Normalize function to ensure both prefix and postfix are normalized, so added this to NormalizeOrg as well